File: visa-dev-samplecode-master/vdp-php/tests/mlc/LocationUpdateTest.php
<?php

namespace Vdp;

class LocationUpdateTest extends \PHPUnit_Framework_TestCase {
	
	public function setUp() {
		$this->conf = parse_ini_file ( "configuration.ini", true );
		$this->visaAPIClient = new VisaAPIClient;
		$strDate = date('Y-m-d\TH:i:s.z\Z', time());
		$this->locationsRequestBody = json_encode ( [
		  "accuracy" => "5000",
		  "cloudNotificationKey" => "03e3ae03-a627-4241-bad6-58f811c18e46",
		  "cloudNotificationProvider" => "1",
		  "deviceId" => $this->conf ['VDP'] ['mlcDeviceId'],
		  "deviceLocationDateTime" => $strDate,
		  "geoLocationCoordinate" => [
		    "latitude" => "37.558546",
		    "longitude" => "-122.271079"
		  ],
		  "header" => [
		    "messageDateTime" => $strDate,
		    "messageId" => $this->conf ['VDP'] ['mlcMessageId']
		  ],
		  "issuerId" => $this->conf ['VDP'] ['mlcIssuerId'],
		  "provider" => "1",
		  "source" => "2"
		] );
	}
	
	public function testlocations() {
		$baseUrl = "mlc/";
		$resourcePath = "locationupdate/v1/locations";
		$statusCode = $this->visaAPIClient->doMutualAuthCall ( 'post', $baseUrl.$resourcePath, 'Location Update Test', $this->locationsRequestBody );
		$this->assertEquals($statusCode, "200");
	}
}

File: vdp-ruby/test/mlc/location_update_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class LocationUpdateTest < Test::Unit::TestCase
  def setup
    @visa_api_client = VisaAPIClient.new
    @strDate = Time.now.utc.strftime("%Y-%m-%dT%H:%M:%S.%LZ")
    @config = YAML.load_file('configuration.yml')
    @locationsRequestBody ='''{
    "accuracy": "5000",
    "cloudNotificationKey": "03e3ae03-a627-4241-bad6-58f811c18e46",
    "cloudNotificationProvider": "1",
    "deviceId": "'''+ @config['mlcDeviceId'] + '''",
    "deviceLocationDateTime": "'''+ @strDate + '''",
    "geoLocationCoordinate": {
      "latitude": "37.558546",
      "longitude": "-122.271079"
    },
    "header": {
      "messageDateTime": "'''+ @strDate + '''",
      "messageId": "'''+ @config['mlcMessageId'] + '''"
    },
    "issuerId": "'''+ @config['mlcIssuerId'] + '''",
    "provider": "1",
    "source": "2"
  }'''
  end
  
  def test_locationUpdate
    base_uri = 'mlc/'
    resource_path = 'locationupdate/v1/locations'
    response_code = @visa_api_client.doMutualAuthRequest("#{base_uri}#{resource_path}", "Location Update test", "post", @locationsRequestBody)
    assert_equal("200", response_code, "Location Update test failed")
  end
end
